src/main/kotlin/skill/TaoQu.kt

```diff
@@ -123,13 +125,28 @@ class TaoQu : MainPhaseSkill() {
             }
             if (r is RobotPlayer) {
                 GameExecutor.post(g, {
-                    val playerAndCard = g.players.flatMap {
-                        if (!it!!.alive || !it.isEnemy(r)) emptyList()
-                        else it.messageCards.mapNotNull { card ->
-                            if (!card.colors.any { c -> c in colors }) null
-                            else PlayerAndCard(it, card)
+                    val color = listOf(Red, Blue, Black).filter {
+                        r.cards.count(it) >= 2
+                    } // 手牌中出现两次的颜色
+                    val players =
+                        g.players.filter { it!!.alive && it !== r && it.messageCards.isNotEmpty() } // 过滤出除了自己且存活有情报的玩家
+                    val moveplayerAndcards = ArrayList<PlayerAndCard>() // 存储可能指定的玩家以及情报牌的集合
+                    var value = Int.MIN_VALUE
+                    for (p in players) {
+                        for (movecard in p!!.messageCards) { //
+                            movecard.colors.any { color.contains(it) } || continue // 遍历到没有任意两张手牌含有相同的颜色跳过
```

Comment:
你这样不写`toList`还是会`java.util.ConcurrentModificationException`
```suggestion
                        for (movecard in p!!.messageCards.filter { c -> c.colors.any { color.contains(it) }) {
                            // 遍历到没有任意两张手牌含有相同的颜色跳过
```